internal/config/config.go

```diff
@@ -0,0 +1,146 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/rancher/wrangler/v3/pkg/kubeconfig"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/rancher/scc-operator/internal/consts"
+	rootLog "github.com/rancher/scc-operator/internal/log"
+	"github.com/rancher/scc-operator/pkg/util/log"
+)
+
+var logger = log.NewComponentLogger("int/config")
+
+// OperatorSettings represents config values that the SCC Operator relies on to run
+// These values are either set by: 1. Reading EnvKey vars, or 2. the ConfigMap used by deployers
+// This ensures that SCC Operator execution remains more uniform regardless of execution context.
+type OperatorSettings struct {
+	OperatorName    string
+	Kubeconfig      string
+	SystemNamespace string
+	LeaseNamespace  string
+	LogFormat       rootLog.Format
+	LogLevel        logrus.Level
+	CattleDevMode   bool
+
+	// DevMode tracks the operators "dev mode" status, when enabled many features will be configured for better dev feedback
+	DevMode               bool
+	DefaultSCCEnvironment consts.SCCEnvironment
+}
+
+// Validate simply validates the configured settings are potentially valid but not if objects exist
+func (s *OperatorSettings) Validate() error {
+	if s.OperatorName == "" {
+		return fmt.Errorf("operator name must be set")
+	}
+	if s.SystemNamespace == "" {
+		return fmt.Errorf("operator must have a valid SCC system namespace")
+	}
+	if s.LeaseNamespace == "" {
+		logger.Warn("operator lease namespace is empty; will default to `kube-system`")
+	}
+	return nil
+}
+
+// Global variable for live configuration.
+var (
+	currentConfig *OperatorSettings
+	mu            sync.RWMutex
+)
+
+// LoadInitialConfig will fetch a value resolver and combine it with a ConfigMap (if exists) to prepare an OperatorSettings
+func LoadInitialConfig(ctx context.Context) (*OperatorSettings, error) {
+	valueResolver := NewValueResolver()
+
+	kubeconfigPath := valueResolver.Get(Kubeconfig, "")
+
+	restKubeConfig, err := kubeconfig.GetNonInteractiveClientConfig(kubeconfigPath).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientSet, err := kubernetes.NewForConfig(restKubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	operatorNamespace := valueResolver.Get(OperatorNamespace, consts.DefaultSCCNamespace)
```

Comment:
Why do we need to pass `consts.DefaultSCCNamespace` here? Can't we use it when declaring `OperatorNamespace` in options.go?